### PR TITLE
feat: check testnet account for funding

### LINKED ISSUE #7: Testnet accounts should be checked for funding when signing in
Following a Testnet reset, when logging into an account that was set up before the reset, the login will fail with a 404 error, since the account is no longer live on the Testnet.

The [`fetchAccount` function](https://github.com/stellar/basic-payment-app/blob/59e4877d04106cf237a59c0abc6b86cfcec4261f/src/lib/stellar/horizonQueries.js#L33) in `/src/lib/stellar/horizonQueries.js` should be modified to go ahead and fund the account in the event it finds a 404 response from the Horizon server.

### PR DESCRIPTION
**Pull request by od-hunter**
fixes #7

### CODE REVIEW

**Comment by od-hunter** [comment]
@ElliotFriend  please review.

File: src/lib/stellar/horizonQueries.js [hunk 1]
```diff
@@ -38,7 +38,16 @@ export async function fetchAccount(publicKey) {
         } catch (err) {
             // @ts-ignore
             if (err.response?.status === 404) {
-                throw error(404, 'account not funded on network')
+                try {
+                    fundWithFriendbot(publicKey)
```
Comment (ElliotFriend) [inline src/lib/stellar/horizonQueries.js]: i think you'll want to `await fundWithFriendbot(...)` here

File: src/lib/stellar/horizonQueries.js [hunk 2]
```diff
@@ -38,7 +38,16 @@ export async function fetchAccount(publicKey) {
...
+
+                    let account = await server.accounts().accountId(publicKey).call()
+                    return account
+                } catch (error) {
```
Comment (ElliotFriend) [inline src/lib/stellar/horizonQueries.js]:
```suggestion
                } catch (err) {
```

This will be more consistent with the rest of the codebase

**Review by ElliotFriend — CHANGES_REQUESTED** [review]
thanks for the PR! i think you're pretty close. i'm requesting a couple nitpicky changes. after they're resolved, I think this will be good to go!

**Comment by od-hunter** [comment]
@ElliotFriend , please review again

File: src/lib/stellar/horizonQueries.js [hunk 3]
```diff
@@ -38,7 +38,16 @@ export async function fetchAccount(publicKey) {
...
             if (err.response?.status === 404) {
-                throw error(404, 'account not funded on network')
+                try {
+                    await fundWithFriendbot(publicKey)
+
+                    let account = await server.accounts().accountId(publicKey).call()
+                    return account
+                } catch (err) {
+                    throw error(500, {
+                        message: `Unable to fund account ${publicKey}: ${error.message}`,
```
Comment (ElliotFriend) [inline src/lib/stellar/horizonQueries.js]:
```suggestion
                        message: `Unable to fund account ${publicKey}: ${err.message}`,
```

this'll need to change to `err`, as well

**Review by ElliotFriend — COMMENTED** [review]
one more tiny nitpick and this is ready to go! Thanks!!

**Comment by od-hunter** [comment]
@ElliotFriend please review again🥲

**Comment by od-hunter** [comment]
Done @ElliotFriend

**Review by ElliotFriend — APPROVED** [review]
excellent! Thanks so much for the contribution!

**Pull request merged**